### ISSUE TITLE
Property value not created with multi-byte character values

### ISSUE THREAD

**Issue by pikanji** [lovata/oc-shopaholic-plugin#138]
In backend product edit screen, if a user enters a value with only multi-byte characters (example: `ほげ`) to the input field for text type property, the property value data is not created. It silently fails to create a property value. It is created when the value includes any single byte characters, like `ほげabc`.

Expected behavior:
* Explicitly fail when it failed to create values.
* Property value is created even with values that has only multi-byte characters.
  
I guess the cause is that it fails to generate a slug with such value with only multi-byte characters.
Though I don't thoroughly understand the necessity for slug of a property value, I hope the property value is created even without its slug.

**Comment by kharanenka** [comment]
Hi! Slug field are used to generate an URL in filtering. How do you generate the URL for catalog page with applied filters? We need to understand what decision to take.
For example, we use:
1. /phones?property[10]=blue|green&page=1&sort=price|asc
2. /phones?property-color-blue-or-green?page=1&sort=price|asc

**Comment by pikanji** [comment]
@kharanenka I think you are talking about slug of Property. I'm guessing that the problem of this issue is caused by generation of slug for each **property value**.

By the way, my answer to your question is below:
Thought I have not thought about filtering with the properties, I generally don't like to use numerical ID.
Numerical ID is not easy for human to identify which data it really is, and inconsistency of the ID between different environments could be sometimes trouble when synching data between those environments.
I guess `[10]` is the ID of property with code `color` in your query examples. So, in this case I prefer using `color`.

I would also try to avoid using unsafe characters in query string, such like `[` and `|`. So, my idea (not tested) is like this;

/phones?property-color=blue,green?page=1&sort=price|asc

**Comment by pikanji** [comment]
@kharanenka Oh, I see.
I usually use [urlencode](http://php.net/manual/en/function.urlencode.php) on Japanese characters that go into URL string. Since it does not encode strings that don't need to be encoded, maybe you can always apply this to generate slug for property value.

**Comment by pikanji** [comment]
HI @kharanenka,
I'm using "Properties for Shopaholic" in production already with the fix below, and it seems working fine. Without this fix, users cannot add Japanese property value.

```diff
--- a/models/PropertyValue.php
+++ b/models/PropertyValue.php
@@ -138,9 +138,9 @@ class PropertyValue extends Model
      */
     public static function getSlugValue($sValue)
     {
-        $sSlug = str_replace([',', '.'], 'x', $sValue);
+        $sSlug = str_replace([',', '.'], 'x', urlencode($sValue));
         $sSlug = Str::slug($sSlug, '');
```

**Comment by kharanenka** [comment]
Fixed in 1.7.3 version. Added settings to help fix the problem with saving of multi-byte character values for PropertyValue model. (Go to Settings -> Shopaholic -> Properties)

Thanks for your contribution.
